x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/hooks/use_rules_list_filter_store.tsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useHistory } from 'react-router-dom';
+import { createKbnUrlStateStorage } from '@kbn/kibana-utils-plugin/public';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import { isEmpty } from 'lodash';
+import { RuleStatus } from '../../../../../common';
+import { RulesListFilters, RulesListProps, UpdateFiltersProps } from '../../../../../types';
+
+type FilterStoreProps = Pick<
+  RulesListProps,
+  | 'lastResponseFilter'
+  | 'lastRunOutcomeFilter'
+  | 'rulesListKey'
+  | 'ruleParamFilter'
+  | 'statusFilter'
+  | 'searchFilter'
+  | 'typeFilter'
+>;
+const RULES_LIST_FILTERS_KEY = 'triggersActionsUi_rulesList';
+
+interface FilterParameters {
+  actionTypes?: string[];
+  lastResponse?: string[];
+  params?: Record<string, string | number | object>;
+  search?: string;
+  status?: RuleStatus[];
+  tags?: string[];
+  type?: string[];
+}
+
+export const convertRulesListFiltersToFilterAttributes = (
+  rulesListFilter: RulesListFilters
+): FilterParameters => {
+  return {
+    actionTypes: rulesListFilter.actionTypes,
+    lastResponse: rulesListFilter.ruleLastRunOutcomes,
+    params: rulesListFilter.ruleParams,
+    search: rulesListFilter.searchText,
+    status: rulesListFilter.ruleStatuses,
+    tags: rulesListFilter.tags,
+    type: rulesListFilter.types,
+  };
+};
+
+export const useRulesListFilterStore = ({
+  lastResponseFilter,
+  lastRunOutcomeFilter,
+  rulesListKey = RULES_LIST_FILTERS_KEY,
+  ruleParamFilter,
+  statusFilter,
+  searchFilter,
+  typeFilter,
+}: FilterStoreProps): {
+  filters: RulesListFilters;
+  setFiltersStore: (params: UpdateFiltersProps) => void;
+  numberOfFiltersStore: number;
+  resetFiltersStore: () => void;
+} => {
+  const history = useHistory();
+  const urlStateStorage = createKbnUrlStateStorage({
+    history,
+    useHash: false,
+    useHashQuery: false,
+  });
+
+  const [rulesListFilterLocal, setRulesListFilterLocal] = useLocalStorage<FilterParameters>(
+    `${RULES_LIST_FILTERS_KEY}_filters`,
+    {}
+  );
+  const hasFilterFromLocalStorage = useMemo(
+    () =>
+      rulesListFilterLocal
+        ? !Object.values(rulesListFilterLocal).every((filters) => isEmpty(filters))
+        : false,
+    [rulesListFilterLocal]
+  );
+
+  const rulesListFilterUrl = useMemo(
+    () => urlStateStorage.get<FilterParameters>('_a') ?? {},
```

Comment:
what does the `_a` mean?

Reply:
that's what o11y use to store all their query parameter.So I keep the same.